python/tests/catalog/test_base.py

```diff
@@ -0,0 +1,410 @@
+            self.__namespaces[namespace].update(updates)
+        else:
+            raise NoSuchNamespaceError(f"Namespace {namespace} not found in the catalog")
+
+
+TEST_TABLE_IDENTIFIER = ("com", "organization", "department", "my_table")
+TEST_TABLE_NAMESPACE = ("com", "organization", "department")
+TEST_TABLE_NAME = "my_table"
+TEST_TABLE_SCHEMA = Schema(schema_id=1)
+TEST_TABLE_LOCATION = "protocol://some/location"
+TEST_TABLE_PARTITION_SPEC = PartitionSpec()
+TEST_TABLE_PROPERTIES = {"key1": "value1", "key2": "value2"}
+
+
+def given_catalog_has_a_table(catalog: InMemoryCatalog) -> Table:
+    return catalog.create_table(
+        identifier=TEST_TABLE_IDENTIFIER,
+        schema=TEST_TABLE_SCHEMA,
+        location=TEST_TABLE_LOCATION,
+        partition_spec=TEST_TABLE_PARTITION_SPEC,
+        properties=TEST_TABLE_PROPERTIES,
+    )
+
+
+def test_namespace_from_tuple():
+    # Given
+    identifier = ("com", "organization", "department", "my_table")
+    # When
+    namespace_from = Catalog.namespace_from(identifier)
+    # Then
+    assert namespace_from == ("com", "organization", "department")
+
+
+def test_namespace_from_str():
+    # Given
+    identifier = "com.organization.department.my_table"
+    # When
+    namespace_from = Catalog.namespace_from(identifier)
+    # Then
+    assert namespace_from == ("com", "organization", "department")
+
+
+def test_name_from_tuple():
+    # Given
+    identifier = ("com", "organization", "department", "my_table")
+    # When
+    name_from = Catalog.table_name_from(identifier)
+    # Then
+    assert name_from == "my_table"
+
+
+def test_name_from_str():
+    # Given
+    identifier = "com.organization.department.my_table"
+    # When
+    name_from = Catalog.table_name_from(identifier)
+    # Then
+    assert name_from == "my_table"
+
+
+def test_create_table(catalog: InMemoryCatalog):
+    table = catalog.create_table(
+        identifier=TEST_TABLE_IDENTIFIER,
+        schema=TEST_TABLE_SCHEMA,
+        location=TEST_TABLE_LOCATION,
+        partition_spec=TEST_TABLE_PARTITION_SPEC,
+        properties=TEST_TABLE_PROPERTIES,
+    )
+    assert catalog.load_table(TEST_TABLE_IDENTIFIER) == table
+
+
+def test_create_table_raises_error_when_table_already_exists(catalog: InMemoryCatalog):
+    # Given
+    given_catalog_has_a_table(catalog)
+    # When
+    with pytest.raises(AlreadyExistsError, match="Table my_table already exists"):
+        catalog.create_table(
+            identifier=TEST_TABLE_IDENTIFIER,
+            schema=TEST_TABLE_SCHEMA,
+        )
+
+
+def test_load_table(catalog: InMemoryCatalog):
+    # Given
+    given_table = given_catalog_has_a_table(catalog)
+    # When
+    table = catalog.load_table(TEST_TABLE_IDENTIFIER)
+    # Then
+    assert table == given_table
+
+
+def test_table_raises_error_on_table_not_found(catalog: InMemoryCatalog):
+    with pytest.raises(NoSuchTableError, match="Table my_table not found"):
```

Comment:
Error messages like this can be confusing if the name is invalid: `Table  not found`. Instead, we prefer to make it clear that the name was included by using a message like `Table does not exist: my_table`. Can you update the error message?